modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/events/SegmentsServicePreAction.java

```diff
@@ -118,7 +118,14 @@ protected void doRun(
 			return;
 		}
 
-		long[] segmentsEntryIds = _getSegmentsEntryIds(
+		long[] segmentsEntryIds = (long[])httpServletRequest.getAttribute(
+			SegmentsWebKeys.SEGMENTS_ENTRY_IDS);
+
+		if (segmentsEntryIds != null) {
+			return;
+		}
+
+		segmentsEntryIds = _getSegmentsEntryIds(
 			httpServletRequest, themeDisplay.getScopeGroupId(),
 			themeDisplay.getUserId());
 
```

Comment:
This parameter was already set in the request here, but we didn't reuse it between ServicePreAction and SegmentsRoleContributor